InkscapeExtension/ExtendedHatchFill.py

```diff
@@ -96,8 +101,109 @@ def recursivelyTraverseSvg(self, a_node_list):
+                     c {w / 3 + r1 * hx}, {h / 3 - r1 * wx} \
+                       {w / 3 - r2 * hx}, {h/ 3 + r2 * wx} \
+                       {x2 - x1 - 2 * od * wx} {y2 - y1 - 2* od * hx }'
+
+         if self.options.mode == "cubic":
+            path = f'M {x1 + od * wx} {y1 + od * hx } \
+                     q {w / 4 + 2 * r1 * hx}, {h / 4 + 2 * r1 * wx} \
+                       {w / 2 + r2 * hx}, {h / 2  + r2 * wx} \
+                     t {(x2 - x1)/ (2) - 2 * od * wx}, {(y2 - y1)/ (2)- 2* od * hx  }'
+
+
+         if self.options.mode == "hatch_line":
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - od * wx) / number_segm
+             dy = (y2 - y1 - od * hx) / number_segm
+             path = f'M {x1 + od * wx}, {y1 + od * hx} l {dx - r1 - od * wx} {dy - r2 - od * hx}'
+             for _ in range(number_segm - 1):
+                 r1 = randint(-cc, cc)
+                 r2 = randint(-cc, cc)
+                 x1 += dx
+                 y1 += dy
+                 path += f'M {x1}, {y1} l {dx - r1} {dy - r2}'
+
+
+         if self.options.mode == "hatch_curv":
+             path = ''
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - 2* od * wx) / number_segm
+             dy = (y2 - y1 - 2 * od * hx) / number_segm
+             for _ in range(number_segm):
+                 r1 = randint(-cc, cc) / 2
+                 r2 = randint(-cc, cc) / 2
+                 path += f'M {x1 + od * wx}, {y1 + od * hx} c {r1} , {r2} {-2 * r1},  {-2 * r2} {dx-1} {dy-1}'
+                 x1 += dx
+                 y1 += dy
+
+         if self.options.mode == "wavy":
+             path = ''
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - 2 * od * wx) / number_segm
+             dy = (y2 - y1 - 2 * od * hx) / number_segm
+             for _ in range(number_segm):
+                 r1 = randint(1, cc)
+                 r2 = randint(-cc, 0)
+                 path += f'M {x1 + od * wx}, {y1 + od * hx} \
+                             c {dx / 3}, {5} \
+                               {dx / 3 * 2},{-5}  \
+                               {dx - od * wx} {dy - od * hx}'
+                 x1 += dx
+                 y1 += dy
+
+
+         if self.options.mode == "oval":
+             r = self.options.hatchSpacing
+             path = ''
+
+             dx = (x2 - x1)
+             dy = (y2 - y1)
+             dxR = r * dx  / totLen
+             dyR = r * dy / totLen
+             number_segm = (totLen / r )
+             for _ in range(int(number_segm)+1):
+                 r1 = randint(1, cc)
+                 r2 = randint(1, cc)
+                 r3 = randint(1, cc) * 2
+                 r4 = randint(1, cc) * 2
+                 r5 = randint(-cc, cc)
+                 r6 = randint(-cc, cc)
+                 r7 = randint(-cc, cc)
+                 r8 = randint(-cc, cc)
+                 path += f'M {x1 }, {y1 } c {r1},{r3}  {-r + r5}, {r/2 + r7} {-r} {0} \
```

Comment:
please change {0} to 0

Reply:
here instead of 0 will be outer contour indent